packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts

```diff
@@ -271,30 +269,50 @@ export class ChainAdapter implements IChainAdapter<ChainTypes.Bitcoin> {
       throw new Error('undefined fee')
 
     // We have to round because coinselect library uses sats per byte which cant be decimals
-    const fast = String(Math.round(feeData.data.fast?.satsPerKiloByte / 1024))
-    const average = String(Math.round(feeData.data.average?.satsPerKiloByte / 1024))
-    const slow = String(Math.round(feeData.data.slow?.satsPerKiloByte / 1024))
+    const fastPerByte = String(Math.round(feeData.data.fast?.satsPerKiloByte / 1024))
+    const averagePerByte = String(Math.round(feeData.data.average?.satsPerKiloByte / 1024))
+    const slowPerByte = String(Math.round(feeData.data.slow?.satsPerKiloByte / 1024))
+
+    const { data: utxos } = await this.providers.http.getUtxos({
+      pubkey
+    })
+
+    type MappedUtxos = Omit<bitcoin.api.Utxo, 'value'> & { value: number }
+    const mappedUtxos: MappedUtxos[] = utxos.map((x) => ({ ...x, value: Number(x.value) }))
```

Comment:
Number is lossy. Is that ok here or should we use BigNumber to maintain accuracy?

Reply:
This number cast scenario is to conform with the `coinselect` interface, so not sure what we can do about that

Reply:
iI believe its fine because thats the format coinSelect expects it. We do it like this in buildSendTransactions for bitcoin and it hasnt been a problem.

they also use a number here in their example:

https://github.com/bitcoinjs/coinselect

Reply:
I still think that we should consistently use the same strategy for holding and passing around numbers (like BigNumber) that avoids loss of precision.

Reply:
So your saying `value` in `GetFeeDataInput` should be a BigNumber? 

I guess I see the point from a consistency standpoint but it needs toe be converted to a number for coinselect and its not being modified in any way before being passed to coinSelect.

Sorry I merged this before i saw this comment